tests/cnf/core/network/metallb/tests/metallb-nodeSelector.go

```diff
@@ -116,6 +118,65 @@ var _ = Describe("MetalLB NodeSelector", Ordered, Label(tsparams.LabelBGPTestCas
 				Expect(err).ToNot(HaveOccurred(), "Failed to collect bgp community status")
 				Expect(len(bgpStatus.Routes)).To(Equal(2))
 			})
+
+		It("Update the node selector option with a label that does not exist",
+			reportxml.ID("61336"), func() {
+				By("Setup test case with services, test pods and bgppeers")
+				frrPod0, frrPod1 := setupTestCase(ipAddressPool, ipAddressPool, frrk8sPods)
+
+				By("Create a bgpadvertisement for BGPPeer1 and IP AddressPool1.")
+				setupBgpAdvertisement(bgpAdvertisementName1, tsparams.NoAdvertiseCommunity, ipaddressPoolName1, 100,
+					[]string{tsparams.BgpPeerName1}, worker0NodeLabel)
+
+				By("Create a bgpadvertisement for BGPPeer2 and IP AddressPool1.")
+				setupBgpAdvertisement(bgpAdvertisementName2, tsparams.CustomCommunity, ipaddressPoolName1,
+					100, []string{tsparams.BgpPeerName2}, worker1NodeLabel)
+
+				By("Verify BGP Establishement and service route advertisement.")
+				verifyBGPConnectivityAndPrefixes(frrPod0, frrPod1, nodeAddrList, addressPool, addressPool)
+
+				By("Create a non existing label to use in test case.")
+				nonExistingLabel := []metav1.LabelSelector{
+					{MatchLabels: map[string]string{corev1.LabelHostname: "non-existing-label"}},
+				}
+
+				By("Update bgpadvertisement2 node selector option with a non existing node label.")
+				updateBgpAdvertisement(bgpAdvertisementName2, nonExistingLabel)
+
+				verifyBGPStatusAndRouteAfterLabelUpdate(frrPod0, frrPod1, removePrefixFromIPList(ipv4NodeAddrList),
+					addressPool)
+			})
+
+		It("Remove from node label used in the node selector option",
+			reportxml.ID("53991"), func() {
+
+				By("Setup test case with services, test pods and bgppeers")
+				frrPod0, frrPod1 := setupTestCase(ipAddressPool, ipAddressPool, frrk8sPods)
+
+				By("Adding test label to compute nodes")
+				addNodeLabel(workerNodeList, metalLbTestsLabel)
+
+				testMetallbNodeLabel := []metav1.LabelSelector{
+					{MatchLabels: metalLbTestsLabel},
+				}
+
+				By("Create a bgpadvertisement for BGPPeer1 and IP AddressPool1.")
+				setupBgpAdvertisement(bgpAdvertisementName1, tsparams.NoAdvertiseCommunity, ipaddressPoolName1,
+					100, []string{tsparams.BgpPeerName1}, worker0NodeLabel)
+
+				By("Create a bgpadvertisement for BGPPeer2 and IP AddressPool1.")
+				setupBgpAdvertisement(bgpAdvertisementName2, tsparams.CustomCommunity, ipaddressPoolName1,
+					100, []string{tsparams.BgpPeerName2}, testMetallbNodeLabel)
+
+				By("Verify BGP Establishement and service route advertisement.")
+				verifyBGPConnectivityAndPrefixes(frrPod0, frrPod1, nodeAddrList, addressPool, addressPool)
+
+				By("Remove custom metallb test label from nodes")
+				removeNodeLabel(workerNodeList, metalLbTestsLabel)
```

Comment:
if the test fails, the label will not removed and will fail other tests. It should be in After as one of the first steps

Reply:
Its part of the test under IT "Remove from node label used in the node selector option"